internal/syscallcompat/sys_darwin.go

```diff
@@ -125,6 +146,72 @@ func MkdiratUser(dirfd int, path string, mode uint32, context *fuse.Context) (er
 	return Mkdirat(dirfd, path, mode)
 }
 
+type attrList struct {
+	bitmapCount uint16
+	_           uint16
+	CommonAttr  uint32
+	VolAttr     uint32
+	DirAttr     uint32
+	FileAttr    uint32
+	Forkattr    uint32
+}
+
+func timesToAttrList(a *time.Time, m *time.Time) (attrList attrList, attributes [2]unix.Timespec) {
+	attrList.bitmapCount = unix.ATTR_BIT_MAP_COUNT
+	attrList.CommonAttr = 0
+	i := 0
+	if m != nil {
+		attributes[i] = unix.Timespec(fuse.UtimeToTimespec(m))
+		attrList.CommonAttr |= unix.ATTR_CMN_MODTIME
+		i += 1
+	}
+	if a != nil {
+		attributes[i] = unix.Timespec(fuse.UtimeToTimespec(a))
+		attrList.CommonAttr |= unix.ATTR_CMN_ACCTIME
+		i += 1
+	}
+	return attrList, attributes
+}
+
+// FutimesNano syscall.
+func FutimesNano(fd int, a *time.Time, m *time.Time) (err error) {
+	attrList, attributes := timesToAttrList(a, m)
+	return fsetattrlist(fd, unsafe.Pointer(&attrList), unsafe.Pointer(&attributes),
+		unsafe.Sizeof(attributes), 0)
+}
+
+// UtimesNanoAtNofollow is like UtimesNanoAt but never follows symlinks.
+//
+// Unfortunately we cannot use unix.UtimesNanoAt since it is broken and just
+// ignores the provided 'dirfd'. In addition, it also lacks handling of 'nil'
+// pointers (used to preserve one of both timestamps).
+func UtimesNanoAtNofollow(dirfd int, path string, a *time.Time, m *time.Time) (err error) {
+	if !filepath.IsAbs(path) {
```

Comment:
Note that I didn't put it in `emulation.go` since `setattrlist` does not exist on Linux. We might want to rework the directory structure anyway, now that it only contains a single function.